### PR TITLE
Symfony 4 upgrade

### PR DESCRIPTION
**Pull request by Shaky212**
This is the upgrade from Symfony 3.4 tot Symfony 4.4. The major steps are:
- resolve deprecations
- upgrade composer to Symfony 4.4
- installl Symfony/flex 

Within Symfony 4 the application and directory structure has changed. The app folder has partly been moved to the config folder en the .env environment. All twig templates have been moved to the templates folder.

Notice the .env contains a parameter administrator_teams used in the dashboard_saml.yaml and checked during compile time in Configuration.php. Previously this was an array, but not supported because of it's dynamic character. The administrator_teams is now a scalar, a string with comma separated values.

See also: https://symfonycasts.com/screencast/symfony4-upgrade/upgrade4

### CODE REVIEW

File: src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Controller/ServiceController.php [hunk 1]
```diff
@@ -165,15 +171,12 @@ public function overviewAction()
...
      * @Security("has_role('ROLE_ADMINISTRATOR')")
-     * @Template()
      *
      * @return RedirectResponse|Response
      */
     public function createAction(Request $request)
     {
         $this->get('session')->getFlashBag()->clear();
-        /** @var LoggerInterface $logger */
-        $logger = $this->get('logger');
```
Comment (MKodde) [inline src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Controller/ServiceController.php:176]: :mechanical_arm: :medal_military:

File: templates/Exception/error.html.twig [hunk 2]
```diff
@@ -1,4 +1,4 @@
-{% extends 'base.html.twig' %}
+{% extends '../../../../../templates/base.html.twig' %}
```
Comment (MKodde) [inline templates/Exception/error.html.twig:1]: Why are the `../` required here. Given the other templates that can simply extend the base template by its name?

**Review by MKodde — COMMENTED** [review]
Awesome work so far! I did note quiet a few issues you might need to address.

For example: 
- Some of the old, unused folders are still in the project `web`, `app`
- Swiftmailer is still showing up in config
- See code review below for more details.

File: docker/Dockerfiledev [hunk 3]
```diff
@@ -1,6 +1,7 @@
 FROM ghcr.io/openconext/openconext-deploy/openconext-core:feature_build_image_with_teams_included AS openconext
 COPY ./conf/prep_oc.sh  /usr/local/sbin/prep_oc.sh
-COPY ./conf/saml20_sp.json /tmp/saml20_sp.json
+#COPY ./conf/saml20_sp.json /tmp/saml20_sp.json
```
[the remark this answers is not in the document]
Comment (MKodde) [reply docker/Dockerfiledev:3]: Also on line 13, we install composer 1 in the `fpm` container. But that conflicts with our new composer packages installed. They require composer 2 to install. Please upgrade to composer 2 :)

**Review by MKodde — CHANGES_REQUESTED** [review]

**Comment by MKodde** [comment]
The jest test still fail as they refer to old, moved fixture files. Should be an easy fix to get a fully green build!

File: docker/Dockerfile [hunk 4]
```diff
@@ -2,9 +2,7 @@ FROM ghcr.io/openconext/openconext-containers/openconext-phpfpm-dev:latest AS no
 COPY . /var/www/html/
 WORKDIR /var/www/html/
 ENV SYMFONY_ENV=prod
-RUN composer install -n --prefer-dist -o --ignore-platform-reqs --no-dev && \
-    composer run-script symfony-scripts
-
+RUN composer install -n --prefer-dist -o --ignore-platform-reqs --no-dev
```
[the remark this answers is not in the document]
Comment (quartje) [reply docker/Dockerfile:5]: Will this install the assets?

**Pull request closed without merging**